### DISCUSSION TITLE
Undefined variable $pagination

### DISCUSSION THREAD

**Discussion by jeyzur** [tighten/jigsaw#704] [Q&A]
Hi everyone, 
I set up a posts page (posts.blade.php) with pagination in the header followed by using $pagination in a foreach loop but after I save, via watch, I keep getting an error that $pagination is undefined. I thought that the pagination key being in the header would make $pagination available. Here's the code:

`---
pagination:
    collection: posts
    perPage: 7
---
foreach($pagination->items as post)
...
`

**Comment by husseinalhammad** [comment]
Hey @jeyzur 

You need to indent `collection` and `perPage`:

```markdown
---
pagination:
  collection: posts
  perPage: 5
---
```
